packages/core/src/lib/persisted.ts

```diff
@@ -0,0 +1,88 @@
+import { writable, type Writable } from 'svelte/store';
+
+const stores: Record<string, Writable<unknown> | undefined> = {};
+
+const getStorage = (type: 'local' | 'session') => {
+  return type === 'local' ? localStorage : sessionStorage;
+};
+
+const createPersisted = <T>(
+  key: string,
+  initialValue: T | null,
+  storage: ReturnType<typeof getStorage>
+) => {
+  const store = writable<T | null>(initialValue, (set) => {
+    const json = storage.getItem(key);
+
+    if (json) {
+      try {
+        set(JSON.parse(json) as T);
+      } catch {
+        set(null);
+      }
+    } else {
+      storage.setItem(key, JSON.stringify(initialValue));
+    }
+
+    const handleStorage = (event: StorageEvent) => {
+      if (event.key === key) {
+        if (event.newValue) {
+          try {
+            set(JSON.parse(event.newValue) as T);
+          } catch {
+            set(null);
+          }
+        } else {
+          set(null);
+        }
+      }
+    };
+
+    window.addEventListener('storage', handleStorage);
+
+    return () => window.removeEventListener('storage', handleStorage);
+  });
+
+  const { subscribe, set } = store;
+
+  return {
+    set(value: T): void {
+      storage.setItem(key, JSON.stringify(value));
```

Comment:
This will raise if the value is not JSON-ifiable. I don't object to the raising, but perhaps we want to raise a more actionable error?

Reply:
Fixed with jsonifiable ✨

Reply:
Oh, also do you mean throwing as well, or do we think TS is enough here?

Reply:
I think TS is enough!